packages/react/src/useChangeEmail.ts

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useMemo } from 'react'
+
+import {
+  ChangeEmailHandlerResult,
+  ChangeEmailOptions,
+  changeEmailPromise,
+  ChangeEmailState,
+  createChangeEmailMachine
+} from '@nhost/core'
+import { useInterpret, useSelector } from '@xstate/react'
+
+import { useNhostClient } from './useNhostClient'
+
+interface ChangeEmailHandler {
+  (email: string, options?: ChangeEmailOptions): Promise<ChangeEmailHandlerResult>
+  /** @deprecated */
+  (email?: unknown, options?: ChangeEmailOptions): Promise<ChangeEmailHandlerResult>
+}
+
+interface ChangeEmailHookResult extends ChangeEmailState {
```

Comment:
Please export this interface. This change should be applied to all the other `Result` interfaces.

My reasoning here is simple: We should make enhancing the behaviour of our hooks as easy to the users of our SDK as possible. By exporting everything that can help this will make our SDK a lot more flexible.